### PR TITLE
Add GH actions script for docker image

### PR DESCRIPTION
**Pull request by hoerup**
# Issue
Related to https://github.com/Project-OSRM/osrm-backend/issues/6134

I'm new to github actions but I think this should do the trick ?

### CODE REVIEW

File: .github/workflows/osrm-backend-docker.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: build and publish container image
+on:
+  create:
+    tags:
+      - v*
```
Comment (mjjbell) [inline .github/workflows/osrm-backend-docker.yml:5]: I think tag filters are only supported on the `push` or `pull_request` events
https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#onpushpull_requestbranchestags

For example, it looks like this triggered when you created a branch with no tag.
https://github.com/hoerup/osrm-backend/actions/runs/1279643123

File: .github/workflows/osrm-backend-docker.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build container image
+        uses: docker/build-push-action@v2
+        with:
+          push: true
+          file: ./docker/Dockerfile
```
Comment (mjjbell) [inline .github/workflows/osrm-backend-docker.yml:26]: To support debug/assertion builds like the old hook, you'll need to pass build-args here
https://github.com/Project-OSRM/osrm-backend/blob/34877482e678e44d995454586191ee39d5c31c4b/docker/hooks/build#L9

File: .github/workflows/osrm-backend-docker.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+          tags: ghcr.io/${{ github.repository }}/osrm-backend:${{ GITHUB_REF#refs/*/ }}
```
Comment (mjjbell) [inline .github/workflows/osrm-backend-docker.yml:27]: Have you considered using [docker/metadata-action](https://github.com/docker/metadata-action)?
If has some nice Docker tag extraction functionality (e.g. we can push `v5.26.0`, and `latest` on the same container).

**Review by mjjbell — CHANGES_REQUESTED** [review]

File: .github/workflows/osrm-backend-docker.yml (hunk 3 above)
Comment (hoerup) [reply .github/workflows/osrm-backend-docker.yml:27]: docker/metadata-action seems nice - but I can't figure out how to combine that with the normal/debug/assertions variations

File: .github/workflows/osrm-backend-docker.yml [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    steps:       
+      - name: Check out the repo
+        uses: actions/checkout@v2
+
+
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v3
+        with:
+          images: ghcr.io/${{ github.repository }}/osrm-backend
```
Comment (mjjbell) [inline .github/workflows/osrm-backend-docker.yml:21]: [github.repository](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context) includes the repository name.

File: .github/workflows/osrm-backend-docker.yml [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+
+      - name: Build container image - normal
+        uses: docker/build-push-action@v2
+        with:
+          push: true
+          file: ./docker/Dockerfile
+          tags:  ${{ steps.meta.outputs.labels }}
+          build-args: 
+          - DOCKER_TAG:  ${{ steps.meta.outputs.labels }}
```
Comment (mjjbell) [inline .github/workflows/osrm-backend-docker.yml:61]: Looks like `build-args` expects a list, not a dictionary.
>  build-args | List | List of build-time variables

With the `github.repository` change, this file should be fork agnostic, you can try building the containers on your own fork to confirm it works.

File: .github/workflows/osrm-backend-docker.yml [hunk 6]
```diff
@@ -0,0 +1,84 @@
+name: build and publish container image
+on:
+  push:
+    tags:
+      - 'v*'
+    branches:
+      - master
```
Comment (mjjbell) [inline .github/workflows/osrm-backend-docker.yml:7]: Hmm, so the `latest` tag will would flip between stable release versions and master builds. I think this is confusing.

**Review by mjjbell — CHANGES_REQUESTED** [review]

**Comment by hoerup** [comment]
I think I got it to where we want:

https://github.com/hoerup/osrm-backend/runs/3743503746?check_suite_focus=true

https://github.com/hoerup/osrm-backend/pkgs/container/osrm-backend

Let me know if you want a clean PR with out the mess

**Pull request closed without merging**